File: src/main/scala/relfections/EventAEnricher.scala
package relfections

import org.springframework.stereotype.Component

/**
  * @author Evgeny Borisov
  */
@Component
class EventAEnricher extends EventEnricher[EventA] {
  override def enrich(event: EventA): EventA = {
    println(event)
    event
  }

  override def myEvents(): Set[Class[_ <: Event]] = {
    Set(classOf[EventA])
  }
}

File: src/main/java/handlers/EventAEnricher.java
package handlers;

import org.springframework.stereotype.Component;

/**
 * @author Evgeny Borisov
 */
@Component
public class EventAEnricher<T extends Event> implements EventEnricher<EventA> {
    @Override
    public Event enrichCurrent(EventA event) {
        System.out.println("handling aaaa");
        System.out.println(event.getAaa());
        return event;
    }
}
